### PR TITLE
Mini Cart template part: give user customized template priority over theme template

### LINKED ISSUE #8929: If a user has modified the Mini Cart template part and the theme introduces a custom one, in the frontend the theme template takes precedence
## To reproduce

1. With TT3, makes some edits to the Mini Cart template part and save it.
2. Verify the edits are visible in the frontend when you open the Mini Cart drawer.
3. Now, let's imagine the theme you are using releases an update which introduces a Mini Cart template part. To do so, create a `mini-cart.html` file in the `parts` subfolder of your theme.
4. In the editor, notice the customized template in step 1 takes priority over the theme one.
5. In the frontend, notice the theme template takes priority over the customized one.

## Expected behavior

I would expect the template part I see in the frontend to be the same I see in the editor. Currently, it might happen that the editor and the frontend show different template parts, which is super-confusing and makes it impossible to introduce changes to the template part.

Regarding which template part should take preference, IMO it should be the user modified one. This seems to be how other WC template work. That means that, in the editor, we are displaying the correct template part, but in the frontend we aren't. 

## Screenshots

Editor | Frontend
--- | ---
imatge | imatge

### PR DESCRIPTION
**Pull request by Aljullu**
Fixes #8929.

#### User Facing Testing

0. With [TT3](https://github.com/WordPress/twentytwentythree/), add the Mini Cart block to your store header via Appearance > Editor.
1. Also in Appearance > Editor, make some edits to the Mini Cart template part and save it.
2. Verify the edits are visible in the frontend when you open the Mini Cart drawer.
3. Now, let's imagine TT3 releases an update which introduces a custom Mini Cart template part. To do so, install this ZIP as a theme (it's TT3 with a `parts/mini-cart.html` file): [twentytwentythree.zip](https://github.com/woocommerce/woocommerce-blocks/files/11200877/twentytwentythree.zip).
4. Verify that going to Appearance > Editor > Template Parts > Mini Cart shows the template you modified in step 1 (instead of the theme template).
5. Verify that in the frontend the Mini Cart drawer shows the template you modified in step 1 (instead of the theme template).

In other words, given that you created a custom template in step 1, you should never see the template from the theme.

Editor | Frontend
--- | ---
imatge | imatge

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Give user customized Mini Cart template part priority over the template part from the theme

### CODE REVIEW

File: src/BlockTemplatesController.php [hunk 1]
```diff
@@ -351,32 +351,7 @@ function( $template ) {
...
-		$check_query         = new \WP_Query( $check_query_args );
-		$saved_woo_templates = $check_query->posts;
-
-		return array_map(
-			function( $saved_woo_template ) {
-				return BlockTemplateUtils::build_template_result_from_post( $saved_woo_template );
-			},
-			$saved_woo_templates
-		);
+		return BlockTemplateUtils::get_block_templates_from_db( $slugs, $template_type );
```
Comment (Aljullu) [inline src/BlockTemplatesController.php:355]: I guess it's ok to move this function to `BlockTemplateUtils.php`?

Comment (albarin) [reply src/BlockTemplatesController.php:355]: Should we use the `BlockTemplateUtils::get_block_templates_from_db` in line 448 and remove this one since it's just wrapping the `BlockTemplateUtils` one?

**Review by albarin — APPROVED** [review]
Pre-approving with just one comment :)

**Comment by Aljullu** [comment]
> Should we use the `BlockTemplateUtils::get_block_templates_from_db` in line 448

Makes sense, done in 02c1ac54e07ca63343ea3a2ab22e30de853b80a5!

> remove this one since it's just wrapping the `BlockTemplateUtils` one?

I'm not sure about this one. Given that this is a public method, another extension could have started using it and we might break it if we remove, no? Do you think we should mark it as deprecated (via [`wc_deprecated_function()`](https://woocommerce.github.io/code-reference/namespaces/default.html#function_wc_deprecated_function))? :thinking:

**Comment by albarin** [comment]
> I'm not sure about this one. Given that this is a public method, another extension could have started using it and we might break it if we remove, no? Do you think we should mark it as deprecated (via [`wc_deprecated_function()`](https://woocommerce.github.io/code-reference/namespaces/default.html#function_wc_deprecated_function))? 🤔

Oh, that makes total sense, I didn't think about that case. We can deprecate it I guess.
